charts/prometheus-node-exporter/templates/service.yaml

```diff
@@ -10,6 +10,11 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+{{- if ge (.Capabilities.KubeVersion.Minor | int) 20 }}
+  ipFamilies:
+{{ toYaml .Values.service.ipFamilies | indent 2 }}
+  ipFamilyPolicy: {{ .Values.service.ipFamilyPolicy }}
+{{- end }}
```

Comment:
Dual-stack services were made available as alpha in release [1.20](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md) behind feature gate `IPv6DualStack`, disabled by default. The condition may not therefore be reliable. The feature became beta in 1.21 and stable in 1.23.

What about a toggle? One would not need to bother with dual stack making thus no change unless it is enabled. For example:

```
spec:
  {{- if .Values.service.ipDualStack.enabled }}
  ipFamilies: {{ toYaml .Values.service.ipDualStack.ipFamilies | nindent 4 }}
  ipFamilyPolicy: {{ .Values.service.ipDualStack.ipFamilyPolicy }}
  {{- end }}
```

Reply:
Great! Done. Add the `flag` for DualStack